eliot-service/eliot/cache_manager.py

```diff
@@ -0,0 +1,366 @@
+                rm_path = pathlib.Path(rm_path)
+                self.total_size -= rm_size
+                rm_path.unlink(missing_ok=True)
+                LOGGER.debug(f"evicted {rm_path} {rm_size:,d}")
+
+            # Since the file exists, add it to the lru
+            LOGGER.debug(f"added {path} {size:,d}")
+            self.lru[lru_key] = size
+
+        LOGGER.debug(f"lru: count {len(self.lru)}, size {self.total_size:,d}")
+
+    def inventory_existing(self):
+        """Walk cache_dir directory and update bookkeeping."""
+        # Reset everything
+        self.total_size = 0
+        self.lru = OrderedDict()
+
+        # Walk directory and update bookkeeping
+        for base, dirs, files in os.walk(str(self.cache_dir)):
+            for fn in files:
+                fn = os.path.join(base, fn)
+                self.update_bookkeeping(pathlib.Path(fn))
+
+    def event_generator(self, nonblocking=False):
+        """Returns a generator of inotify events."""
+        if nonblocking:
+            # NOTE(willkg): Timeout of 0 should return immediately if there's nothing
+            # there
+            timeout = 0
+        else:
+            timeout = 1000
+
+        inotify = inotify_simple.INotify(nonblocking=nonblocking)
+        watch_flags = (
+            flags.ACCESS
+            | flags.CREATE
+            | flags.DELETE
+            | flags.MODIFY
+            | flags.MOVED_FROM
+            | flags.MOVED_TO
+        )
+        watch_descriptor = inotify.add_watch(str(self.cache_dir), watch_flags)
+
+        LOGGER.info("Entering loop")
+        try:
+            while True:
+                for event in inotify.read(timeout=timeout):
+                    LOGGER.debug(event)
+                    event_flags = flags.from_mask(event.mask)
+                    # Ignore changes to directories
+                    if flags.ISDIR in event_flags:
+                        continue
+
+                    path = self.cache_dir / event.name
+                    if flags.CREATE in event_flags:
+                        self.process_create(path, event)
+                    elif flags.MOVED_FROM in event_flags:
+                        self.process_moved_from(path, event)
+                    elif flags.MOVED_TO in event_flags:
+                        self.process_moved_to(path, event)
+                    elif flags.MODIFY in event_flags:
+                        self.process_modify(path, event)
+                    elif flags.DELETE in event_flags:
+                        self.process_delete(path, event)
+                    elif flags.ACCESS in event_flags:
+                        self.process_access(path, event)
+                    else:
+                        LOGGER.debug(f"ignored {path} {event}")
+
+                yield
+
+        finally:
+            inotify.rm_watch(watch_descriptor)
+
+    def run_loop(self):
+        """Run cache manager in a loop."""
+        for event in self.event_generator():
+            pass
```

Comment:
`self.event_generator()` doesn't yield events, so this is confusing and should get fixed.